Model/articles.model.js

```diff
@@ -13,17 +13,36 @@ exports.selectArticle = (articleID) => {
     })
 }
 
-exports.selectAllArticles = () => {
-    return db.query(
-        `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count FROM articles
+exports.selectAllArticles = (topic) => {
+   
+    let sqlQuery = 
+    `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count FROM articles
         LEFT JOIN comments
-        ON comments.article_id = articles.article_id
+        ON comments.article_id = articles.article_id `
+
+    if(topic){
+        return db.query(
+        sqlQuery += 
+        `WHERE topic = $1
         GROUP BY articles.article_id
+        ORDER BY created_at DESC`, [topic]
+        )
+        .then(({rows, rowCount}) => {
+            if(rowCount === 0){
```

Comment:
What about a topic that exists but has no articles? This needs to be accounted for in yours tests and code.